gitlint/git.py

```diff
@@ -54,10 +54,11 @@ class GitCommit(object):
         In the context of gitlint, only the git context and commit message are required.
     """
 
-    def __init__(self, context, message, date=None, author_name=None, author_email=None, parents=None,
+    def __init__(self, context, message, sha, date=None, author_name=None, author_email=None, parents=None,
```

Comment:
I think we should make ```sha``` an optional argument because we won't always have the sha available. For example, gitlint supports just piping arbitrary text into it for linting, in this case we don't have the sha (that's the reason why all these other parameters are also optional)